### ISSUE TITLE
Add a "Paragragh" style and use it for paragraph instead of the default "Text_20_body"

### ISSUE THREAD

**Issue by QiangF** [kjambunathan/org-mode-ox-odt#250]
I know we have "OrgUser", "Standard" and the "Text_20_body". "Text_20_body" is used in a lot of places, like caption, table cell text.

The issue is that by default in China, paragraph has 2 char indent, it's stupid I know, but if I add that indent to "Text_20_body", it ruins table cell text and maybe list indent.

So I am suggesting add a "Paragraph" style that has 2 char indent if you prefer. And make a change in org-odt-paragraph.

	    ;; Case 4: If an element has no parent (i.e., root node),
	    ;; use the fallback style "Paragraph".
	    (cl-loop for el in data
		     ;; Fallback style.
		     with style = (or
				   ;; Style set in `org-odt-link--inline-image'
				   ;; or `org-odt-link--inline-formula'.
				   (get-text-property 0 :p-style contents)
				   "Paragraph")      ; @my-modification

**Comment by kjambunathan** [comment]
I think this has come up before ...

Let me what is a clean way to approach it

**Comment by QiangF** [comment]
Indeed, I have raised a similar issue before 
https://github.com/kjambunathan/org-mode-ox-odt/pull/59

I think it will do no harm to add a Paragraph style.

**Comment by kjambunathan** [comment]
Can you post a ODT file with malformed indentation? I would like to see a screenshot and corresponding source code.

The screenshot etc. will serve as reminders to our future selves ... on what the original problem was.

Layout problems are better understood with screenshots, than loads of description.

------


I am more concerned about the list styles, and their interaction with paragraph styles.

> Indeed, I have raised a similar issue before https://github.com/kjambunathan/org-mode-ox-odt/pull/59

> I think it will do no harm to add a Paragraph style.

Instead of touching the 'Text_20_body', I may introduce 'OrgListBody' or 'OrgTableBody' or some such thing.

Btw, `Text_20_body' will not occur in normal tables. (They occur only in List Tables and Transcluded Tables)

------

For example, see   [Text Body style with intent affects bulleted list style as well ](https://ask.libreoffice.org/t/text-body-style-with-intent-affects-bulleted-list-style-as-well/27158) and also [[O] ox-odt: List items contain body paragraphs, confusing styles](https://lists.gnu.org/archive/html/emacs-orgmode/2017-08/msg00391.html).  I have filed a libreoffice bug report [LO Bug 141539](https://bugs.documentfoundation.org/show_bug.cgi?id=141539) - First Line Indent doesn't work for paragraphs within a list item ([edit](https://bugs.documentfoundation.org/show_bug.cgi?id=141539#)).  The LO Bug has been acknowledged by Regina.  In her remarks, she notes that the list styles are "old styles".  I don't know if switching the list style to "new list styles" will fix the problem.

This involves lots of experimentation ... and that is one of the reasons it has been around for a while.

Yes, I acknowledge that there is a problem ... and IMO, the problem has to be fixed in the list styles.  (The fact that paragraph and list styles interact in complex and buggy ways makes things harder still)

I also may want to move away from 'Legacy List Styles' to modern 'List Styles' at some in time.

**Comment by QiangF** [comment]
You are right! If I add "fo:text-indent="0.8cm"" to "Text_20_body", I got the first line indent in list-table:

    hello world

    #+caption: table
    #+label: tbl:1
    #+attr_odt: :style "GriddedTable" :rel-width 100
    |r1c1|r1c2|

    1. list
    2. list

    #+attr_odt: :list-table t
    -
      -
        #+caption: org
        #+label: 1
        #+attr_odt: :width 7.1 :style "Caption"
        [[./org-mode-unicorn.png]]
      -
        #+caption: org
        #+label: 2
        #+attr_odt: :width 7.1 :style "Caption"
        [[./org-mode-unicorn.png]]


    #+attr_odt: :list-table t
    -
      -
        1. list
        2. list
      -
        1. list
        2. list
      -
        list table

![img_2023_09_18__14_45_57](https://github.com/kjambunathan/org-mode-ox-odt/assets/8192731/a3e0b90e-00a3-4cdc-b49d-a58eb457cc9f)

Creating a new style like 'OrgListBody' or 'OrgTableBody' will be a better solution.
